fixed errors, added quest initiator

JayMan1 opened the pull request.

RobertMcDermot commented on lib/entities/npcs/animals/piggy.dart:
```diff
@@ -125,6 +125,8 @@ class Piggy extends NPC {
 		SkillManager.learn(SKILL, email);
 		nibbleCounts[email] = (nibbleCounts[email] ?? 0) + 1;
 
+		QuestEndpoint.questLogCache[email].offerQuest('Q11');
```
Can you add a comment here and say what quest this will be starting? The way we have the quest files organized, it is hard to glance at what Q11 associates with.

RobertMcDermot commented on lib/entities/npcs/animals/piggy.dart:
```diff
@@ -125,6 +125,8 @@ class Piggy extends NPC {
 		SkillManager.learn(SKILL, email);
 		nibbleCounts[email] = (nibbleCounts[email] ?? 0) + 1;
 
```
You'll need to add a line here to actually make this event count towards their active quest log:

messageBus.publish(new RequirementProgress('piggyNibble', email));

This basically sends a message which will be received by a quest listener for the user matching "email". It doesn't much matter what you call "piggyNibble" so long as it's a unique string among all quest requirements and that it matches (via regex) the eventType in the quest json.

RobertMcDermot commented on lib/quests/json/piggy_nibbler.json:
```diff
@@ -0,0 +1,106 @@
+  "requirements": [
+    {
+      "id": "Q11R2",
+      "type": "counter_unique",
```
counter_unique is meant to only increase the count for unique evenTypes. For example, in the tree petter quest, we want to only give the player credit if they are petting different kinds of trees. So our eventType is treePet.* which means it will match things like "treePetBubble" and "treePetFruit". But since the type is "counter_unique" we will only increase the count if the player has never before had a certain type of event.

In this piggy quest case, I think you want to use "counter" for the type so that it will increase the count each time they nibble a piggy. I also think you can just change eventType to "piggyNibble" to match my above comment in the piggy.dart file.

RobertMcDermot commented on lib/quests/json/piggy_nibbler.json:
```diff
@@ -0,0 +1,106 @@
+  "description": "Go nibble 6 pigs",
+  "requirements": [
+    {
+      "id": "Q11R2",
```
This will work fine, but we could also just change it to Q11R1. The only requirement here is that the id is unique among all quests. So usually I just created a string from the quest id and the requirement number. Since this is the first requirement in this quest, I would tend to call it R1.

RobertMcDermot commented on lib/entities/npcs/animals/piggy.dart:
```diff
@@ -124,6 +124,8 @@ class Piggy extends NPC {
 		SkillManager.learn(SKILL, email);
 		nibbleCounts[email] = (nibbleCounts[email] ?? 0) + 1;
+		messageBus.publish(new RequirementProgress('piggyNibble', email));
+		QuestEndpoint.questLogCache[email].offerQuest('Q11, Piggy Nibbler Quest');
```
I had just meant a comment in the code about which quest this was so that people who are reading it know. This call is going to search for a quest with an id of "Q11, Piggy Nibbler Quest" but you're defining it as "Q11" so this is going to fail.

RobertMcDermot commented on lib/quests/json/piggy_nibbler.json:
```diff
@@ -0,0 +1,106 @@
+      "id": "Q11R1",
+      "type": "counter",
+      "numRequired": 6,
+      "eventType": "piggyNibble.*",
```
This will work, but it would be better if it was just "piggyNibble"

JayMan1 commented:
Ok fixed those things as well. I apologize, it's been a little while since
I've worked with code. I'm not sure if I added the new comment correctly,
since I'm now seeing lines 128 and 129 highlighted in red. so if I didn't I
apologize yet again.

On Fri, Sep 16, 2016 at 9:42 AM, Robert McDermot <notifications@github.com>
wrote:

> *@RobertMcDermot* requested changes on this pull request.
>
> Added just a few comments
> ------------------------------
>
> In lib/entities/npcs/animals/piggy.dart
> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>
> :
>
> > @@ -124,6 +124,8 @@ class Piggy extends NPC {
>  		StatManager.add(email, Stat.piggies_nibbled);
>  		SkillManager.learn(SKILL, email);
>  		nibbleCounts[email] = (nibbleCounts[email] ?? 0) + 1;
> +		messageBus.publish(new RequirementProgress('piggyNibble', email));
> +		QuestEndpoint.questLogCache[email].offerQuest('Q11, Piggy Nibbler Quest');
>
> I had just meant a comment in the code about which quest this was so that
> people who are reading it know. This call is going to search for a quest
> with an id of "Q11, Piggy Nibbler Quest" but you're defining it as "Q11" so
> this is going to fail.
> ------------------------------
>
> In lib/quests/json/piggy_nibbler.json
> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>
> :
>
> > @@ -0,0 +1,106 @@
> +{
> +  "id": "Q11",
> +  "title": "Meat Galore",
> +  "description": "Go nibble 6 pigs",
> +  "requirements": [
> +    {
> +      "id": "Q11R1",
> +      "type": "counter",
> +      "numRequired": 6,
> +      "eventType": "piggyNibble.*",
>
> This will work, but it would be better if it was just "piggyNibble"
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUSyBovJ7lCoUyPb9wRYOb9VsDdVoRewks5qqpzNgaJpZM4J95lX>
> .
>

JayMan1 commented:
Oh wait, I think that's just where I deleted things. So disregard the
highlighted in red comment.

On Fri, Sep 16, 2016 at 12:42 PM, itsaluminum <itsaluminum1217@gmail.com>
wrote:

> Ok fixed those things as well. I apologize, it's been a little while since
> I've worked with code. I'm not sure if I added the new comment correctly,
> since I'm now seeing lines 128 and 129 highlighted in red. so if I didn't I
> apologize yet again.
>
> On Fri, Sep 16, 2016 at 9:42 AM, Robert McDermot <notifications@github.com
> > wrote:
>
>> *@RobertMcDermot* requested changes on this pull request.
>>
>> Added just a few comments
>> ------------------------------
>>
>> In lib/entities/npcs/animals/piggy.dart
>> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>
>> :
>>
>> > @@ -124,6 +124,8 @@ class Piggy extends NPC {
>>  		StatManager.add(email, Stat.piggies_nibbled);
>>  		SkillManager.learn(SKILL, email);
>>  		nibbleCounts[email] = (nibbleCounts[email] ?? 0) + 1;
>> +		messageBus.publish(new RequirementProgress('piggyNibble', email));
>> +		QuestEndpoint.questLogCache[email].offerQuest('Q11, Piggy Nibbler Quest');
>>
>> I had just meant a comment in the code about which quest this was so that
>> people who are reading it know. This call is going to search for a quest
>> with an id of "Q11, Piggy Nibbler Quest" but you're defining it as "Q11" so
>> this is going to fail.
>> ------------------------------
>>
>> In lib/quests/json/piggy_nibbler.json
>> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>
>> :
>>
>> > @@ -0,0 +1,106 @@
>> +{
>> +  "id": "Q11",
>> +  "title": "Meat Galore",
>> +  "description": "Go nibble 6 pigs",
>> +  "requirements": [
>> +    {
>> +      "id": "Q11R1",
>> +      "type": "counter",
>> +      "numRequired": 6,
>> +      "eventType": "piggyNibble.*",
>>
>> This will work, but it would be better if it was just "piggyNibble"
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ChildrenOfUr/coUserver/pull/39#pullrequestreview-354670>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AUSyBovJ7lCoUyPb9wRYOb9VsDdVoRewks5qqpzNgaJpZM4J95lX>
>> .
>>
>
>

RobertMcDermot commented on lib/entities/npcs/animals/piggy.dart:
```diff
@@ -124,7 +124,10 @@ class Piggy extends NPC {
-
+		messageBus.publish(new RequirementProgress('piggyNibble', email));
+		QuestEndpoint.questLogCache[email].offerQuest('Q11');
+		//Piggy Nibbler Quest
```
FYI: generally the comment will go before the line that it is referring to. I'm going to go ahead and pull this anyway so that we can test out the changes.
